cpp/README.md

```diff
@@ -237,6 +237,23 @@ add_mpi_test(
 Define `${PROJECT}_TEST_FORMATTING:BOOL` CMake variable to enforce formatting during
 the `test` make target.
 
+##### List C++ files formatted with ClangFormat
+
+Here are the steps used to establish the list of files to format:
+1. read file `compile_commands.json` generated by CMake to collect all C++
+   compilation units.
+1. compile every compilation unit with the `-M` flag to get the list
+   of included C++ headers.
+1. filter out every file that fulfill one of the following condition:
+   * is not inside `${CMAKE_SOURCE_DIR}` directory
+   * is inside `${CMAKE_BINARY_DIR}` directory
+   * is inside special directories used by some CI systems:
+     * `${CMAKE_SOURCE_DIR}/HOME`
+     * `${CMAKE_SOURCE_DIR}/INSTALL`
+   * its absolute path matches one of the regular expression specified
+     in `${PROJECT}_ClangFormat_EXCLUDES_RE` CMake variable
+1. Filter out files according to the `${PROJECT}_FORMATTING_ON` CMake variable
+
```

Comment:
LGTM.

I will add note here about what we faced today in CoreNEURON project:
* Under GitHub Actions CI, there was no boost installed. So tests targets were not added and hence compiler_commands.json didn't include files from tests directory. Hence these files were never checked.
* Adding a brief note saying "if you are integrating coding convention into your project, make sure to enable necessary targets so that all files are included in the build and hence as part of `clang-format` target`....." might save some debugging time / surprises?

Reply:
> Under GitHub Actions CI, there was no boost installed. So tests targets were not added and hence compiler_commands.json didn't include files from tests directory. Hence these files were never checked.

IMO, this makes me think we should rather use something like `git ls-files` to obtain a list of files to be checked, and skip calling into Python to do all the processing.

Reply:
I don't know much about `git ls-files` but something robust (or straightforward) would be helpful. @alkino  and myself had confusion until @tristan0x explained how it works.